apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -219,6 +238,15 @@ export const CreateHookSheet = ({
         },
       }
     )
+
+    // grant permissions to new function
+    if (values.postgresValues.statements.length > 0) {
+      await executeSql({
```

Comment:
This doesn't run after `updateAuthConfig`, runs in parallel because there's no await on it. You can move this code into `onSuccess` and add something in `onError` which reverses the revokation.